File: app/src/main/java/org/michaelbel/moviemade/ui/modules/reviews/ReviewsActivity.java
package org.michaelbel.moviemade.ui.modules.reviews;

import android.os.Bundle;

import org.michaelbel.moviemade.R;
import org.michaelbel.moviemade.ui.base.BaseActivity;

import androidx.appcompat.widget.AppCompatTextView;
import androidx.appcompat.widget.Toolbar;
import butterknife.BindView;
import butterknife.ButterKnife;

public class ReviewsActivity extends BaseActivity {

    public int movieId;
    private ReviewsFragment fragment;

    @BindView(R.id.toolbar) Toolbar toolbar;
    @BindView(R.id.toolbar_title) AppCompatTextView toolbarTitle;
    @BindView(R.id.toolbar_subtitle) AppCompatTextView toolbarSubtitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reviews);
        ButterKnife.bind(this);

        movieId = getIntent().getIntExtra("id", 0);
        String movieTitle = getIntent().getStringExtra("title");

        setSupportActionBar(toolbar);
        toolbar.setNavigationIcon(R.drawable.ic_arrow_back);
        toolbar.setNavigationOnClickListener(v -> finish());
        toolbar.setOnClickListener(v -> fragment.recyclerView.smoothScrollToPosition(0));

        toolbarTitle.setText(R.string.reviews);
        toolbarSubtitle.setText(movieTitle);

        fragment = (ReviewsFragment) getSupportFragmentManager().findFragmentById(R.id.fragment);
        if (fragment != null) {
            fragment.presenter.loadReviews(movieId);
        }
    }
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/features/reviews/activity/ReviewsActivity.kt
package org.michaelbel.moviemade.presentation.features.reviews.activity

import android.os.Bundle
import kotlinx.android.synthetic.main.activity_default.*
import org.michaelbel.moviemade.R
import org.michaelbel.moviemade.core.entity.Movie
import org.michaelbel.moviemade.core.utils.EXTRA_MOVIE
import org.michaelbel.moviemade.presentation.base.BaseActivity
import org.michaelbel.moviemade.presentation.features.reviews.fragment.ReviewsFragment

class ReviewsActivity: BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_default)

        val movie = intent.getSerializableExtra(EXTRA_MOVIE) as Movie

        setSupportActionBar(toolbar)
        toolbar.setNavigationOnClickListener { finish() }
        supportActionBar?.setTitle(R.string.reviews)
        supportActionBar?.subtitle = movie.title

        if (savedInstanceState == null) {
            startFragment(ReviewsFragment.newInstance(movie), container.id)
        }
    }
}